src/vm/i386/asmhelpers.S

```diff
@@ -1162,20 +1162,30 @@ NESTED_END ResolveWorkerChainLookupAsmStub, _TEXT
 NESTED_ENTRY BackPatchWorkerAsmStub, _TEXT, NoHandler
     PROLOG_BEG
     PROLOG_PUSH eax // it may contain siteAddrForRegisterIndirect
+    PROLOG_PUSH ebx
     PROLOG_PUSH ecx
     PROLOG_PUSH edx
     PROLOG_END
 
-    sub     esp, 4     //  for 16 bytes align
+    // Align Stack Frame
+    mov     ebx, esp
+    and     esp, -16
+
+    // Call BackPatchWorkerStaticStub
+    sub     esp, 8     //  16-bytes align padding
```

Comment:
So it looks like a JIT codegen issue. Could you please disass the function at frame 3 (you'll need to use x/i for that and find the function address range manually) so that we can confirm that?